lib/webhooks/__tests__/register.test.ts

```diff
@@ -0,0 +1,309 @@
+
+interface RegisterTestWebhook {
+  topic: string;
+  handler: WebhookHandler;
+  checkMockResponse?: MockResponse;
+  responses?: MockResponse[];
+}
+
+interface RegisterTestResponse {
+  topic: string;
+  registerReturn: RegisterReturn;
+  expectedSuccess?: boolean;
+  responses: MockResponse[];
+}
+
+interface MutationParams {
+  [key: string]: any;
+}
+
+const REGISTER_PARAMS = {
+  accessToken: 'some token',
+  shop: 'shop1.myshopify.io',
+};
+
+describe('shopify.webhooks.register', () => {
+  it('sends a post request to the given shop domain with the webhook data as a GraphQL query in the body and the access token in the headers', async () => {
+    const topic = 'PRODUCTS_CREATE';
+    const handler = HTTP_HANDLER;
+    const responses = [mockResponses.successResponse];
+
+    const registerReturn = await registerWebhook({topic, handler, responses});
+
+    assertWebhookRegistrationRequest('webhookSubscriptionCreate', {
+      topic,
+      callbackUrl: `"https://test_host_name/webhooks"`,
+    });
+    assertRegisterResponse({registerReturn, topic, responses});
+  });
+
+  it('returns a result with success set to false, body set to empty object, when the server doesn’t return a webhookSubscriptionCreate field', async () => {
+    const topic = 'PRODUCTS_CREATE';
+    const handler = HTTP_HANDLER;
+    const responses = [mockResponses.httpFailResponse];
+
+    const registerReturn = await registerWebhook({topic, handler, responses});
+
+    assertWebhookRegistrationRequest('webhookSubscriptionCreate', {
+      topic,
+      callbackUrl: `"https://test_host_name/webhooks"`,
+    });
+    assertRegisterResponse({
+      registerReturn,
+      topic,
+      responses,
+      expectedSuccess: false,
+    });
+  });
+
+  it('sends an eventbridge registration GraphQL query for an eventbridge webhook registration', async () => {
+    const topic = 'PRODUCTS_CREATE';
+    const handler = EVENT_BRIDGE_HANDLER;
+    const responses = [mockResponses.eventBridgeSuccessResponse];
+
+    const registerReturn = await registerWebhook({topic, handler, responses});
+
+    assertWebhookRegistrationRequest('eventBridgeWebhookSubscriptionCreate', {
+      topic,
+      arn: '"arn:test"',
+    });
+    assertRegisterResponse({registerReturn, topic, responses});
+  });
+
+  it('sends a pubsub registration GraphQL query for a pubsub webhook registration', async () => {
+    const topic = 'PRODUCTS_CREATE';
+    const handler = PUB_SUB_HANDLER;
+    const responses = [mockResponses.pubSubSuccessResponse];
+
+    const registerReturn = await registerWebhook({topic, handler, responses});
+
+    assertWebhookRegistrationRequest('pubSubWebhookSubscriptionCreate', {
+      topic,
+      pubSubProject: '"my-project-id"',
+      pubSubTopic: '"my-topic-id"',
+    });
+    assertRegisterResponse({registerReturn, topic, responses});
+  });
+
+  it('deletes and recreates a webhook if the path changes', async () => {
```

Comment:
Nit: the order in `register` is create first then delete ... I initially though the order of the `responses` was incorrect.
```suggestion
  it('creates a new webhook registration if the path changes, then deletes the old one', async () => {
```